mithril-aggregator/src/tools/url_sanitizer.rs

```diff
@@ -1,10 +1,67 @@
 use anyhow::{anyhow, Context};
 use reqwest::Url;
+use std::fmt::{Display, Formatter};
+use std::ops::Deref;
 
 use mithril_common::StdResult;
 
+/// A sanitized URL, guaranteed to have a trailing slash and no empty segments
+#[derive(Debug, Clone, PartialOrd, PartialEq, Eq, Ord, Hash)]
+pub struct SanitizedUrl {
+    internal_url: Url,
+}
+
+impl SanitizedUrl {
```

Comment:
This structure represent a Uri that is not a resource. 
We may named it to reflect that. 
`SanitizedUrl` doesn't mean much. This gives the impression of having a valid Url

Reply:
We share this concern but could not find a name that would fit that description.

Since we could not find anything better we renamed it to `SanitizedUrlWithTrailingSlash` with hope we found a better name in the future.